Add LINQPad plugin

kirill-gerasimenko opened the pull request:
Hi,

Here is PR that adds LinqPad view plugin. I find it very useful. 
Hope it will be for others.

Commit:
Add LINQPad plugin

Commit:
Change net version in nuget config

Commit:
Restore nuget packages on build for Linqpad plugin project

Commit:
Add Linpad dependency to Test.FsEye packages.config

stephen-swensen commented:
This is really nice! But it doesn't compile in VS2010... are you able / willing to back port?

kirill-gerasimenko commented:
Actually I don't have VS2010 on any of my machines :( What I did here is taken VS2015 project format, lowered ToolsVersion to 4.0 and hoped for the best :)
At my another branch I've converted all the projects to the VS2015 format and latest F#, but also changed folders structure (and did some other additions). But I'm not sure you are interested in this kind of changes.

stephen-swensen commented:
All cool, I think I should have a pretty easy time fixing the project for VS2010 myself -- I am still trying to maintain backwards compatibility there.

stephen-swensen commented:
So I've got the project format fixed for VS2010 and pushed to a branch for now: https://github.com/SwensenSoftware/fseye/tree/kirill-gerasimenko-linqpad_plugin. But

- I want to test that the plugin works on mono
- I am concerned about the size of the linqpad .exe, and haven't looked into the license terms for distributing that .exe

Ultimately, I think this plugin may be a nice candidate for distribution through a separate channel rather than packaged as a default plugin (e.g. distribute as a separate NuGet package). One way or another, this is a really nice plugin and would be great to make widely available.

kirill-gerasimenko commented:
Hi,

If the plugin is not included in the main distribution than it means that it should be manually copied to "plugins" folder along with it's dependencies, right? This doesn't sound easy :)
I can extend initialization mechanism (which is now opening module and adding print transformer) with a module's variable which holds additional list of folders to be probed during start, something like (I haven't looked at initialization yet, it's just an example of the idea): 
```fsharp
open Swensen.FsEye.Fsi
Swensen.FsEye.PluginsFolders.add @"c:\..."
fsi.AddPrintTransformer eye.Listener //attached the listener
```

What do you think?
